### PR TITLE
Online DDL: `ALTER VIEW` to respect `--postpone-completion` strategy flag

### PR DESCRIPTION
**Pull request by shlomi-noach**
## Description

Addresses https://github.com/vitessio/vitess/issues/11898, see bug description there.

This PR makes `ALTER VIEW` respect `--postpone-completion` strategy flag, in a forward migration as well as in a revert migration.

Added `endtoend` tests to validate.

## Related Issue(s)

https://github.com/vitessio/vitess/issues/11898

## Checklist

-   [ ] "Backport to:" labels have been added if this change should be back-ported
-   [x] Tests were added or are not required
-   [x] Documentation was added or is not required

### CODE REVIEW

**Review by iheanyi — APPROVED** [review]
LGTM 👀

**Review by no-itsbackpack — APPROVED** [review]

File: go/vt/vttablet/onlineddl/executor.go [hunk 1]
```diff
@@ -2145,25 +2145,36 @@ func (e *Executor) scheduleNextMigration(ctx context.Context) error {
...
 					return err
 				}
 			}
 		}
-		if ddlAction == sqlparser.AlterStr || !postponeCompletion {
+
+		if !(isImmediateDDL && postponeCompletion) {
 			// Any non-postponed migration can be scheduled
-			// postponed ALTER can be scheduled
+			// postponed ALTER ALTER can be scheduled (because gh-ost or vreplication will postpone the cut-over)
```
Comment (deepthi) [inline go/vt/vttablet/onlineddl/executor.go:2177]: typo: ALTER is repeated.

**Review by deepthi — APPROVED** [review]

File: go/vt/vttablet/onlineddl/executor.go (hunk 1 above)
Comment (shlomi-noach) [reply go/vt/vttablet/onlineddl/executor.go:2177]: fixed

**Comment by shlomi-noach** [comment]
Replace and superseded by https://github.com/vitessio/vitess/pull/11910, which also handles INSTANT ddl as fast range partition rotation.

**Pull request closed without merging**